iconsdk/signed_transaction.py

```diff
@@ -17,32 +17,39 @@
 from hashlib import sha3_256
 
 from iconsdk.builder.transaction_builder import Transaction
+from iconsdk.exception import DataTypeException
 from iconsdk.libs.serializer import serialize
 from iconsdk.utils import get_timestamp
-from iconsdk.utils.convert_type import convert_int_to_hex_str, convert_params_value_to_hex_str
-from iconsdk.utils.hexadecimal import add_0x_prefix
+from iconsdk.utils.convert_type import convert_int_to_hex_str
+from iconsdk.utils.gen_tx_data import generate_data_value
 from iconsdk.wallet.wallet import Wallet
 
 
 class SignedTransaction:
 
-    def __init__(self, transaction: Transaction, wallet: Wallet):
+    def __init__(self, transaction: Transaction, wallet: Wallet, step_limit: int = None):
         """Converts raw transaction into the signed transaction object having a signature.
 
         :param transaction: A transaction object not having a signature field yet
         :param wallet: A wallet object
         """
+        if transaction.step_limit is None:
+            if step_limit is None:
```

Comment:
This line should be checked first before `if transcation.step_limit is None`, since the step_limit value of transaction should be overridden by the given `step_limit`.

Reply:
I applied to the code. Tnx!!